src/schematools/datasetcollection.py

```diff
@@ -24,29 +24,23 @@ class DatasetCollection(metaclass=Singleton):
     """
 
     def __init__(self) -> None:
-        """Initialize the DatasetCollection.
-
-        Args:
-            schema_loader: An alternative schema loader can be provided
-                If schema_loader is None, the default url loader is used.
-        """
+        """Initialize the DatasetCollection."""
         self.datasets_cache: Dict[str, DatasetSchema] = {}
-        self.schema_loader: Optional[loaders.SchemaLoader] = None
+        self._schema_loader: Optional[loaders.SchemaLoader] = None
 
     def set_schema_loader(self, schema_loader: loaders.SchemaLoader) -> None:
```

Comment:
waarom niet `@schema_loader.setter` op de property?

Reply:
Goed plan. De eerste stap de `@property` was al gedaan, dan is dit het logische vervolg.